src/MetricsTimer.cpp

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "MetricsTimer.h"
+#include "CURLReporter.h"
+#include "CommandHandler.h"
+#include "Converter.h"
+
+#ifdef _MSC_VER
+std::vector<char> getHostname() {
+  std::vector<char> Hostname;
+  return Hostname;
+}
+#else
+#include <unistd.h>
+std::vector<char> getHostname() {
+  std::vector<char> Hostname;
+  Hostname.resize(256);
+  gethostname(Hostname.data(), Hostname.size());
+  if (Hostname.back() != 0) {
+    // likely an error
+    Hostname.back() = 0;
+  }
+  return Hostname;
+}
+#endif
+
+namespace Forwarder {
+
+void MetricsTimer::start() {
+  Logger->trace("Starting the MetricsTimer");
+  Running = true;
+  AsioTimer.async_wait(
+      [this](std::error_code const & /*error*/) { this->reportStats(); });
+  TimerThread = std::thread(&MetricsTimer::run, this);
+}
+
+void MetricsTimer::waitForStop() {
+  Logger->trace("Stopping MetricsTimer");
+  Running = false;
+  AsioTimer.cancel();
+  TimerThread.join();
+}
+
+std::unique_lock<std::mutex> MetricsTimer::get_lock_converters() {
+  return std::unique_lock<std::mutex>(converters_mutex);
+}
+
+void MetricsTimer::reportStats() {
+  KafkaInstanceSet->logStats();
+  auto m1 = g__total_msgs_to_kafka.load();
```

Comment:
Can we have the names be a bit more descriptive instead of using `m1`, `b1` etc.?